battleship-service/src/main/java/dreamteam/battleship/service/springcontroller/gamecontroller/GameControllerBuilder.java

```diff
@@ -1,20 +1,29 @@
 package dreamteam.battleship.service.springcontroller.gamecontroller;
 
 import dreamteam.battleship.logic.movement.MovementManager;
+import dreamteam.battleship.service.springcontroller.model.GameMode;
 import dreamteam.battleship.service.springcontroller.model.Player;
 
 /**
  * Created by ehsan on 09.08.16.
  */
 public class GameControllerBuilder {
 
-    public static final IGameController gameControllerInstance(Player player, MovementManager manager, boolean isSalute){
-        if(isSalute){
-            // TODO salute controller
-            return new NormalController(player, manager);
-        }else {
-            return new NormalController(player, manager);
+    public static final IGameController gameControllerInstance(Player player, MovementManager manager, GameMode mode){
+        IGameController controller = null;
+        switch (mode) {
```

Comment:
Maybe you can use EnumMap<GameMode, IGameController> instead of this switch case in order to follow Open/Closed principle?